compiler/src/dmd/file_manager.d

```diff
@@ -175,121 +175,86 @@ nothrow:
 
         whichPathFoundThis = -1;
 
-        // see if we should check for the module locally.
-        bool checkLocal = pathCache.pathExists(filename);
-        const sdi = FileName.forceExt(filename, hdr_ext);
-        if (checkLocal && FileName.exists(sdi) == 1)
-            return sdi;
-        scope(exit) FileName.free(sdi.ptr);
-
-        const sd = FileName.forceExt(filename, mars_ext);
-        if (checkLocal && FileName.exists(sd) == 1)
-            return sd;
-        scope(exit) FileName.free(sd.ptr);
-
-        if (checkLocal)
+        // List of extensions to match, in order of precedence.
+        const(char)[][4] extensions = [
+            FileName.forceExt(filename, hdr_ext),
+            FileName.forceExt(filename, mars_ext),
+            FileName.forceExt(filename, i_ext),
+            FileName.forceExt(filename, c_ext),
+        ];
```

Comment:
Yet to confirm, but the thinking is that it would now be possible to just tack on `h_ext` here, and not have any of the downsides caused by locally found C headers overriding D modules in the `-Isrc` path.